pkg/schemapatcher/patch_gen.go

```diff
@@ -0,0 +1,29 @@
+package schemapatcher
+
+import (
+	"os"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	kyaml "sigs.k8s.io/yaml"
+)
+
+var RequiredFeatureSet = os.Getenv("OPENSHIFT_REQUIRED_FEATURESET")
+
+// mayHandleFile returns true if this manifest should progress past the file collection stage.
+// Currently, the only check is the feature-set annotation.
+func mayHandleFile(filename string, rawContent []byte) bool {
+	manifest := &unstructured.Unstructured{}
+	if err := kyaml.Unmarshal(rawContent, &manifest); err != nil {
+		return true
+	}
+
+	manifestFeatureSet := manifest.GetAnnotations()["release.openshift.io/feature-set"]
```

Comment:
Can `GetAnnotations` ever return a nil?

Reply:
>Can GetAnnotations ever return a nil?

Yes, it can.  You can read values from a nil map: https://go.dev/play/p/hHg_-G6RpKE